FirebaseRemoteConfig/Sources/RCNUserDefaultsManager.m

```diff
@@ -141,6 +142,21 @@ - (void)setLastActiveTemplateVersion:(NSString *)templateVersion {
   }
 }
 
+- (NSMutableDictionary<NSString *, NSString *> *)customSignals {
```

Comment:
Could we make this immutable instead? There's only a small window of time where the custom signals needs to be mutated. We could make a mutable copy, make the change, and then pass on an immutable copy for archiving.

Reply:
Thanks, I have made it immutable. UserDefaults stores immutable copies, even if a mutable map is passed. When updating UserDefaults, we are already creating its mutable copy, so it works as expected.